prometheus-collectors/class-multisite-stats-collector.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Automattic\VIP\Prometheus;
+
+use Prometheus\Gauge;
+use Prometheus\RegistryInterface;
+
+/**
+ * @codeCoverageIgnore
+ */
+class Multisite_Stats_Collector implements CollectorInterface {
+	private ?Gauge $network_site_gauge = null;
+
+	public function initialize( RegistryInterface $registry ): void {
+		if ( is_multisite() ) {
+			$this->network_site_gauge = $registry->getOrRegisterGauge(
+				'site',
+				'count',
+				'Number of network sites',
+				[ 'status' ]
+			);
+		}
+	}
+
+	public function collect_metrics(): void {
+		if ( ! $this->network_site_gauge ) {
+			return;
+		}
+
+		$sites_count = wp_count_sites();
```

Comment:
Better limit the request context on where this runs, maybe do on admin requests only and also utilize object cache + https://developer.wordpress.org/reference/functions/wp_cache_get_last_changed/ and use multisite create/delete hooks